### DISCUSSION TITLE
Linux help - Can't edit config file once I run portainer/docker compost

### DISCUSSION THREAD

**Discussion by Jonnehs** [blakeblackshear/frigate#3629] [Ask A Question]
Honestly I have about a thousand questions, some based on my very basic Linux knowledge, but the main being

I can't seem to edit the source config file once I've run frigate from docker, even if I stop it the file is locked to read only

I can't find the path i need to give it in the portainer config to load my USB Coral, /dev/bus/usb has a 001 and 002 entry but neither work, but I can see the coral on the base ubuntu OS?

**Comment by NickM-27** [comment]
> I can't seem to edit the source config file once I've run frigate from docker, even if I stop it the file is locked to read only

How are you tried to edit it? (through what software). Odds are it is some config somewhere. You could use something like visual studio code in docker which works great and can be setup to have automatic config validation for the frigate config.

> I can't find the path i need to give it in the portainer config to load my USB Coral, /dev/bus/usb has a 001 and 002 entry but neither work, but I can see the coral on the base ubuntu OS?

You can just pass in the entire usb bus and frigate will find the correct USB. Your setup is a bit confusing to me, but if you aren't running docker directly on ubuntu then you'll need to make sure the USB bus that has the coral is passed through every layer up to frigate.

**Reply by Jonnehs** [reply]
Thanks, I think i solved the read-only config thing..

For the USB Coral, I have tried the docker config;

    devices:
      - /dev/bus/usb/001/:/dev/bus/usb

I've tried simple /dev/bus/usb and usb/002 or 001 which is what i see under /dev/bus/usb, but none of those work. But I do see the TPU in the Ubunti base OS

jonneh@cctv01:~/frigate$ lsusb
Bus 002 Device 002: ID 1a6e:089a Global Unichip Corp.

however frigate just kills itself saying none is present

[2022-08-11 15:53:22] frigate.edgetpu                INFO    : Attempting to load TPU as usb,
[2022-08-11 15:53:47] frigate.edgetpu                ERROR   : No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors.,
Process detector:coral:,
Traceback (most recent call last):,
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 160, in load_delegate,
    delegate = Delegate(library, options),
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 119, in __init__,
    raise ValueError(capture.message),
ValueError,
,
During handling of the above exception, another exception occurred:,
,
Traceback (most recent call last):,
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap,
    self.run(),
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run,
    self._target(*self._args, **self._kwargs),
  File "/opt/frigate/frigate/edgetpu.py", line 136, in run_detector,
    object_detector = LocalObjectDetector(,
  File "/opt/frigate/frigate/edgetpu.py", line 44, in __init__,
    edge_tpu_delegate = load_delegate("libedgetpu.so.1.0", device_config),
  File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 162, in load_delegate,
    raise ValueError('Failed to load delegate from {}\n{}'.format(,
ValueError: Failed to load delegate from libedgetpu.so.1.0,
,

**Reply by NickM-27** [reply]
Okay a couple things here. Definitely follow the docs and just pass the device as `- /dev/bus/usb:/dev/bus/usb`

also, it seems your TPU is not being initialized. It shows up as `Global Unicorp Chip` and then once it initializes it will show up as `Google Inc.`. You may want to try a different USB cable or different USB port on your device to ensure it is getting the power it needs to startup successfully.

When I had my USB it would show up as `Global Unicorp Chip` and frigate would fail to start the first time, but then the coral would initialize and frigate would see it as expected.
